src/main/java/make_post/drivers/MakePostDatabaseAccess.java

```diff
@@ -0,0 +1,139 @@
+package make_post.drivers;
+
+import com.opencsv.exceptions.CsvException;
+import make_post.use_case.MakePostDsGateway;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.CSVWriter;
+import make_post.use_case.make_post_exceptions.MakePostException;
+
+import java.io.*;
+import java.util.*;
+
+public class MakePostDatabaseAccess implements MakePostDsGateway {
+    private final String filepath;
+    public MakePostDatabaseAccess(String filepath){
+        this.filepath = filepath;
+
+    }
+    @Override
+    public int getNumberOfPosts(){
+        String filePath = filepath + "numPostsCreated.csv";
+        File file = new File(filePath);
+        try {
+            // Create an object of fileReader
+            // class with CSV file as a parameter.
+            FileReader fileReader = new FileReader(file);
+
+            // create csvReader object passing
+            // file reader as a parameter
+            CSVReader csvReader = new CSVReaderBuilder(fileReader).withSkipLines(1).build();
+            int numPostsCreated = Integer.parseInt(csvReader.peek()[0]);
+            csvReader.close();
+            return numPostsCreated;
+        } catch (IOException e) {
+            System.out.println("Either wrong path or file has not been formatted correctly. ");
+            return 0;
+        }
+    }
+
+    @Override
+    public void setNumberOfPosts(int newNumPostsCreated) {
+        String filePath = filepath + "numPostsCreated.csv";
+        File file = new File(filePath);
+
+        try {
+            FileReader fileReader = new FileReader(file);
+            // create FileWriter object with file as parameter
+            CSVReader csvReader = new CSVReader(fileReader);
+            // create CSVWriter object filewriter object as parameter
+            List<String[]> csvBody = csvReader.readAll();
+            csvBody.get(1)[0] = String.valueOf(newNumPostsCreated);
+            for(int i = 0; i < csvBody.size(); i++){
+                if(i >= csvBody.size()){
+                    break;
+                }
+                if(csvBody.get(i).length == 0 && csvBody.get(i)[0].equals("")){
+                    csvBody.remove(i);
```

Comment:
I think mutating while iterating over a mutable object is dangerous. I think an example list `csvBody = ["hi", ,]` will cause the for loop to not perform its intended task (the resulting list is `csvBody = ["hi", ]` which still contains an empty entry). I think a better choice would be to just create a new list and add the non-empty rows to the new list.

Reply:
Ok I'll update this!